sdks/java/core/src/main/java/org/apache/beam/sdk/schemas/FieldAccessDescriptor.java

```diff
@@ -300,6 +340,16 @@ public FieldAccessDescriptor withNestedField(
     return withNestedField(field, fieldAccess);
   }
 
+  public FieldAccessDescriptor withNestedFieldAs(
+      String nestedFieldName, String nestedFieldRename, FieldAccessDescriptor fieldAccess) {
+    FieldDescriptor field =
+        FieldDescriptor.builder()
+            .setFieldName(nestedFieldName)
+            .setFieldRename(nestedFieldRename)
+            .build();
+    return withNestedField(field, fieldAccess);
+  }
+
```

Comment:
Consider adding a test for this in `FieldAccessDescriptorTest`, and adding a docstring